subgraphs/_reference_/src/sdk/protocols/generic/index.ts

```diff
@@ -0,0 +1,147 @@
+import { BIGINT_ZERO, ZERO_ADDRESS } from "../../util/constants";
+
+export class CustomEventType {
+  block: ethereum.Block;
+  transaction: ethereum.Transaction;
+  logIndex: BigInt;
+  event: ethereum.Event | null;
+
+  constructor() {
+    this.block = new ethereum.Block(
+      Bytes.empty(),
+      Bytes.empty(),
+      Bytes.empty(),
+      Address.fromString(ZERO_ADDRESS),
+      Bytes.empty(),
+      Bytes.empty(),
+      Bytes.empty(),
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      null,
+      null
+    );
+    this.transaction = new ethereum.Transaction(
+      Bytes.empty(),
+      BIGINT_ZERO,
+      Address.fromString(ZERO_ADDRESS),
+      null,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      Bytes.empty(),
+      BIGINT_ZERO
+    );
+    this.logIndex = BIGINT_ZERO;
+    this.event = null;
+  }
+
+  static initialize(
+    block: ethereum.Block,
+    transaction: ethereum.Transaction,
+    logIndex: BigInt,
+    event: ethereum.Event | null = null
+  ): CustomEventType {
+    const customEvent = new CustomEventType();
+    customEvent.block = block;
+    customEvent.transaction = transaction;
+    customEvent.logIndex = logIndex;
+    customEvent.event = event;
+
+    return customEvent;
+  }
+}
+
+export class SDK {
+  ProtocolManager: ProtocolManager;
+  PoolsManager: PoolsManager;
+  EventsManager: EventsManager;
+  AccountsManager: AccountsManager;
+  TokensManager: TokensManager;
+  Pricer: TokenPricer;
+  event: CustomEventType;
+
+  constructor(
+    config: Config,
+    pricer: TokenPricer,
+    tokenInitializer: TokenInitializer,
+    event: CustomEventType
+  ) {
+    this.ProtocolManager = new ProtocolManager(config, event);
+    this.TokensManager = new TokensManager(
+      this.ProtocolManager,
+      pricer,
+      tokenInitializer
+    );
+    this.AccountsManager = new AccountsManager(this.ProtocolManager);
+    this.PoolsManager = new PoolsManager(this.ProtocolManager);
+    this.EventsManager = new EventsManager(
+      this.PoolsManager,
+      this.TokensManager
+    );
+    this.event = event;
+  }
+
+  static initialize<T>(
+    config: Config,
+    pricer: TokenPricer,
+    tokenInitializer: TokenInitializer,
+    event: T
+  ): SDK {
+    if (event instanceof ethereum.Event) {
+      const customEvent = CustomEventType.initialize(
+        event.block,
+        event.transaction,
+        event.logIndex,
+        event
+      );
+      return new SDK(config, pricer, tokenInitializer, customEvent);
+    }
+    if (event instanceof ethereum.Call) {
+      const customEvent = CustomEventType.initialize(
+        event.block,
+        event.transaction,
+        BIGINT_ZERO
+      );
+      return new SDK(config, pricer, tokenInitializer, customEvent);
+    }
+    return new SDK(config, pricer, tokenInitializer, new CustomEventType());
+  }
+
+  create_pool() {
+    this.TokensManager;
+  }
+
+  update_balances(
```

Comment:
Do you want to expose this directly at the SDK level instead of at the PoolManager?

What will be the purpose of the PoolManager?

Reply:
Updated this to be more like the bridges